src/main.rs

```diff
@@ -313,7 +313,7 @@ async fn main() -> Result<()> {
         if is_reboot || is_power_off || is_reload_config {
             loop {
                 {
-                    let retain_flag = retain_flag.lock().unwrap();
+                    let retain_flag = retain_flag.lock().expect("Mutex lock is safe");
```

Comment:
Could you explain why this is safe?

Reply:
If the message in expect about why the `retain_flag` is always safe needs to change, please also change the message about the `running_flag` in the `retain_periodically` function as well.